__tests__/app.test.js

```diff
@@ -101,4 +101,100 @@ describe("/api/articles", () => {
         });
     });
   });
+  describe("GET article comments by id", () => {
+    test("200 + returns array of comments with correct properties and values", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const comments = body.comments;
+          expect(comments).toBeSortedBy("created_at", { descending: true });
+          expect(comments.length).toBeGreaterThan(0);
+          comments.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                article_id: expect.any(Number),
+              })
+            );
+          });
+        });
+    });
+    test("200 + return no comments when article has no comments", () => {
+      return request(app)
+        .get("/api/articles/4/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const comments = body.comments;
+          expect(comments).toHaveLength(0);
+        });
+    });
+    test("404 + return msg when article isn't found", () => {
+      return request(app)
+        .get("/api/articles/9999/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body).toEqual({ msg: "Not Found" });
+        });
+    });
+    test("400 + return msg when passed invalid id", () => {
+      return request(app)
+        .get("/api/articles/banana/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body).toEqual({ msg: "Bad Request" });
+        });
+    });
+  });
+
+  describe("POST comments by article id", () => {
+    test("201 when passed valid comment to valid article id", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({
+          username: "lurker",
+          body: "hello",
+        })
+        .expect(201);
```

Comment:
You have a comment coming back that you could test the properties for. This would be more thorough

Reply:
More info below...